module/plugins/captcha/ReCaptcha.py

```diff
@@ -354,3 +356,45 @@ def _challenge_v2(self, key, secure_token=None):
             self.fail(_("Recaptcha max retries exceeded"))
 
         return result, challenge
+
+    # solve interactive captcha (javascript required), use when non-JS captcha fallback for v2 is not allowed
+    def _challenge_v2_js(self, key):
+        self.log_debug("challenge ReCaptcha v2 interactive")
+        captchaManager = self.pyload.captchaManager
+
+        try:
+            time_ref = ("%.2f" % time.time())[-6:].replace(".", "")
+            filename = "captcha_request_%s_%s.nofile" % (self.pyfile.plugin.__name__, time_ref)
+            # we don't have an actual image to decrypt -> pass the URL and the site-key for further use in plugins/hooks
+            img = {'url': self.pyfile.url, 'sitekey': self.retrieve_key(None)}
+            self.task = captchaManager.newTask(img, "interactive", filename , "interactive")
+
+            captchaManager.handleCaptcha(self.task)
+
+            # @TODO: Move to `CaptchaManager` in 0.4.10
+            self.task.setWaiting(120)
+            while self.task.isWaiting():
+                self.pyfile.plugin.check_status()
+                time.sleep(1)
+
+        finally:
+            captchaManager.removeTask(self.task)
+
+        result = self.task.result
+
+        if self.task.error:
```

Comment:
This code is copied from Captcha.py. Can you instead create a method `check_captcha_result` or something in Captcha.py and then use that method in Captcha.decrypt_image and here, so that we do not duplicate the code.

Reply:
Maybe the whole code of the _challenge_v2_js function should be moved to Captcha.py to a decrypt_interactive (or something like this) function, to share the same code with decrypt_image there. And in ReCaptcha.py we need only to call the new function.

That's a design decision.... maybe more people can give their opinion on this.

Reply:
Captcha.py is the base class and should only provide general functionality to handle captchas, while ReCaptcha.py is more specific. Captcha.py should not contain anything that is specific to ReCaptcha only, like "sitekey".

Reply:
@pyblub Look at the changes I pushed to the 'all ' branch.
ReCaptcha will first try the old `fallback` method and if fails will try `2js` version (I renamed v2_javascript to 2js).
The advantages of this are:
It is more general version, no need to change the hoster to explicitly use 2js.
It can save 9kw credits because it will use 9kw's recaptchav2 only when necessary.